src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1498 @@
+
+        if (asprintf(&j->object_path, "/org/freedesktop/sysupdate1/job/_%" PRIu32, j->id) < 0)
+                return -ENOMEM;
+
+        r = hashmap_put(t->manager->jobs, UINT32_TO_PTR(j->id), j);
+        if (r < 0)
+                return r;
+
+        t->manager->current_job_id = j->id;
+
+        *ret = TAKE_PTR(j);
+        return 0;
+}
+
+static int job_parse_stdout(int _fd, JsonVariant **ret) {
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        _cleanup_close_ int fd = _fd;
+        _cleanup_fclose_ FILE *f = NULL;
+        struct stat st;
+        int r;
+
+        assert(ret);
+
+        if (fstat(fd, &st) < 0)
+                return log_error_errno(errno, "Failed to stat stdout fd: %m");
+
+        assert(S_ISREG(st.st_mode));
+
+        if (st.st_size == 0)
+                return 0;
+
+        if (lseek(fd, SEEK_SET, 0) == (off_t) -1)
+                return log_error_errno(errno, "Failed to seek to beginning of memfd: %m");
+
+        f = take_fdopen(&fd, "r");
+        if (!f)
+                return log_error_errno(errno, "Failed to reopen memfd: %m");
+
+        r = json_parse_file(f, "stdout", 0, &v, NULL, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse JSON: %m");
+
+        *ret = TAKE_PTR(v);
+        return 0;
+}
+
+static int job_on_pid(sd_event_source *s, const siginfo_t *si, void *userdata) {
+        Job *j = ASSERT_PTR(userdata);
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(json_variant_unrefp) JsonVariant *json = NULL;
+        int r;
+
+        assert(s);
+        assert(si);
+
+        assert(j->target->busy);
+        j->target->busy = false;
+
+        j->pid = 0;
+
+        if (si->si_code != CLD_EXITED) {
+                assert(IN_SET(si->si_code, CLD_KILLED, CLD_DUMPED));
+                sd_bus_error_setf(&error, SD_BUS_ERROR_FAILED,
+                                  "Job terminated abnormally with signal %s.",
+                                  signal_to_string(si->si_status));
+        } else if (si->si_status != EXIT_SUCCESS)
+                sd_bus_error_setf(&error, SD_BUS_ERROR_FAILED,
+                                  "Job failed with exit code %i.",
+                                  si->si_status);
+        else {
+                r = job_parse_stdout(TAKE_FD(j->stdout_fd), &json);
+                if (r < 0)
+                        sd_bus_error_set_errnof(&error, r, "Failed to parse JSON: %m");
+        }
+
+        r = sd_bus_emit_signal(
+                        j->manager->bus,
+                        "/org/freedesktop/sysupdate1",
+                        "org.freedesktop.sysupdate1.Manager",
+                        "JobRemoved",
+                        "uoi",
+                        j->id,
+                        j->object_path,
+                        si->si_status);
+        if (r < 0)
+                log_error_errno(r, "Cannot emit JobRemoved message, ignoring: %m");
```

Comment:
if we ignore errors, please never log beyond LOG_WARNING. i.e. use `log_warrning_errno()` here